### PR TITLE
Added some window functions and unit tests

### PR DESCRIPTION
**Pull request by Risto-Stevcev**

### CODE REVIEW

**Comment by garyb** [comment]
Thanks for working on these. I'll come back to this later once `Aff` has been updated (for the tests).

**Comment by garyb** [comment]
`Aff` can be bumped to 2.0.0 now, which hopefully will get things passing again.

**Comment by Risto-Stevcev** [comment]
@garyb Got the checks to pass

File: src/DOM/HTML/Window.purs [hunk 1]
```diff
@@ -13,3 +42,71 @@ foreign import location :: forall eff. Window -> Eff (dom :: DOM | eff) Location
 foreign import innerWidth :: forall eff. Window -> Eff (dom :: DOM | eff) Int
 
 foreign import innerHeight :: forall eff. Window -> Eff (dom :: DOM | eff) Int
+
+foreign import alert :: forall eff. Window -> String -> Eff (alert :: ALERT | eff) Unit
+
+foreign import close :: forall eff. Window -> Eff (window :: WINDOW | eff) Unit
+
+foreign import closed :: forall eff. Window -> Eff (window :: WINDOW | eff) Boolean
```
Comment (garyb) [inline src/DOM/HTML/Window.purs]: Please remove this, it's a nonstandard property - this library is intentionally only implementing standards, even when things are commonly available.

File: src/DOM/HTML/Window.purs [hunk 2]
```diff
@@ -13,3 +42,71 @@ foreign import location :: forall eff. Window -> Eff (dom :: DOM | eff) Location
...
+
+foreign import confirm :: forall eff. Window -> String -> Eff (confirm :: CONFIRM | eff) Boolean
+
+foreign import length :: forall eff. Window -> Eff (window :: WINDOW | eff) Int
+
+foreign import minimize :: forall eff. Window -> Eff (window :: WINDOW | eff) Unit
```
Comment (garyb) [inline src/DOM/HTML/Window.purs]: Please remove this one too.

File: src/DOM/HTML/Window.purs [hunk 3]
```diff
@@ -13,3 +42,71 @@ foreign import location :: forall eff. Window -> Eff (dom :: DOM | eff) Location
...
+
+foreign import _open
+  :: forall eff
+   . (forall a. a -> Maybe a)
+  -> (forall a. Maybe a)
+  -> Window
+  -> String
+  -> String
+  -> String
+  -> Eff (window :: WINDOW | eff) (Maybe Window)
```
Comment (garyb) [inline src/DOM/HTML/Window.purs]: You could implement this with `Nullable` instead and then use `toMaybe`. Something like this:

https://github.com/purescript-contrib/purescript-dom/blob/f7b786d9f005df0a41b1da63d3525b8d8ae55457/src/DOM/HTML/Event/DragEvent/DataTransfer.purs#L20-L23

The FFI code can just be currying then, with no need to handle the `Nothing` / `Just` arguments.

File: src/DOM/HTML/Window.purs [hunk 4]
```diff
@@ -13,3 +42,71 @@ foreign import location :: forall eff. Window -> Eff (dom :: DOM | eff) Location
...
+prompt :: forall eff. Window -> String -> Eff (prompt :: PROMPT | eff) (Maybe String)
+prompt = _prompt Just Nothing
+
+foreign import _prompt
+  :: forall eff
+   . (forall a. a -> Maybe a)
+  -> (forall a. Maybe a)
+  -> Window
+  -> String
+  -> Eff (prompt :: PROMPT | eff) (Maybe String)
```
Comment (garyb) [inline src/DOM/HTML/Window.purs]: Another case where `Nullable` can be used.

**Review by garyb — CHANGES_REQUESTED** [review]
Thanks for working on these! There's a few things I'd like removed or updated before merging though.

**Comment by Risto-Stevcev** [comment]
@garyb Updated it

How do you feel about extending purescript-nullable to have `Undefinable` and `NullOrUndefinable`? `Nullable` is a lot easier to use than `Data.Foreign.Null`, but it doesn't have the equivalent `Data.Foreign.Undefined` and `Data.Foreign.NullOrUndefined`

**Comment by garyb** [comment]
`Nullable` is already `NullOrUndefined`, as it's not intended for precise data format matching like `Foreign` is - it's just a helper for when you have functions that might not return a value. Is the distinction necessary here? I wouldn't have though so.

**Review by garyb — CHANGES_REQUESTED** [review]
Sorry, just noticed another thing - the argument orders are all "javascripty" rather than "purescripty" - `window` should be the last argument in each case, as then you can do things like:

``` purescript
alert "hi" =<< window
```

**Comment by Risto-Stevcev** [comment]
Rearranged it

**Review by garyb — APPROVED** [review]

**Comment by garyb** [comment]
Thanks!

**Pull request merged**